src/parsers/forecast_parser.jl

```diff
@@ -108,42 +62,57 @@ Returns:
     A PowerSystems forecast stuct array
 """
 
-function make_forecast_array(sys::System,ts_dict::Dict)
-    ts_map = _retrieve(ts_dict, Union{TimeSeries.TimeArray,DataFrames.DataFrame},Dict(),[]) #find key-path to timeseries data fields
-    fc = Array{Forecast}(undef, 0)
-    all_devices = collect(get_components(Component,sys))
-    for (key,val) in ts_map
-        ts = _access(ts_dict,vcat(val,key)) #retrieve timeseries data
-        if (typeof(ts)==DataFrames.DataFrame) & (size(ts,2) > 2)
-            devices = [d for d in all_devices if d.name in string.(names(ts))]
-            for d in devices
-                dd = isa(d,LoadZones) ? d.buses : [d]
+function make_forecast_array(sys::System, parsed_forecasts::Vector{Dict})
+    return make_forecast_array(get_components(Component, sys))
+end
+
+function make_forecast_array(all_components, parsed_forecasts::Vector{Dict})
+    # TODO this code could be cleaner if the incoming data was only DataFrame or TimeArray
+    forecasts = Vector{Forecast}()
+    for forecast in parsed_forecasts
+        data = forecast["data"]
+        if data isa DataFrames.DataFrame && size(data, 2) > 2
```

Comment:
I think this is trying to find if the DataFrame has a forecast for more than one device. Looking at the RTS data each forecast CSV contains data for all the devices of the same type.

Reply:
I think that I didn’t understand time arrays when this was written. We could probably replace the data frame route with a timearray

Reply:
I think that I didn’t understand time arrays when this was written. We could probably replace the data frame route with a timearray